torch/csrc/jit/script/init.cpp

```diff
@@ -168,6 +168,19 @@ struct VISIBILITY_HIDDEN PythonValue : public SugaredValue {
     return ss.str();
   }
 
+  std::vector<std::shared_ptr<SugaredValue>> asTuple(
+      const SourceRange& loc,
+      Method& m,
+      const c10::optional<size_t>& size_hint = {}) override {
+    const std::string type_str = typeString(self);
+    std::stringstream ss;
+    ss << kind() << " cannot be used as a tuple";
+    if (type_str == "ModuleList" || type_str == "Sequential") {
```

Comment:
You can do instanceof directly (instead of comparing type strings) with something like py::isinstance(self, py::module(“torch.nn”).attr(“ModuleList”)